subsys/bluetooth/controller/ll_sw/ull_adv.c

```diff
@@ -1420,17 +1420,19 @@ uint8_t ull_adv_data_set(struct ll_adv_set *adv, uint8_t len,
 	struct pdu_adv *pdu;
 	uint8_t idx;
 
-	/* Dont update data if directed or extended advertising. */
+	/* Dont update data if directed */
 	prev = lll_adv_data_peek(&adv->lll);
-	if ((prev->type == PDU_ADV_TYPE_DIRECT_IND) ||
-	    (IS_ENABLED(CONFIG_BT_CTLR_ADV_EXT) &&
-	     (prev->type == PDU_ADV_TYPE_EXT_IND))) {
+	if (prev->type == PDU_ADV_TYPE_DIRECT_IND) {
 		/* TODO: remember data, to be used if type is changed using
 		 * parameter set function ll_adv_params_set afterwards.
 		 */
 		return 0;
 	}
 
+	if (len > 31) {
+		return BT_HCI_ERR_INVALID_PARAM;
+	}
```

Comment:
I think the data length should be a define. I'm assuming that the same function will be used when setting ext adv data, and it would be better if this length check value was depending on a define, or at least so that extended advertising data above 31 octets can be set.

Reply:
this code is only used to set data for legacy sets either via legacy command or as a fallback from ext command so no need for conditional check

as for define, smth like this in `pdu.h`?
```
/* Advertisement channel maximum legacy advertising/scan data size */
#define PDU_AC_DATA_SIZE_MAX 31
```

Reply:
Yes something like that. I'm guessing it's a bit out of reach of the controller to use the `BT_GAP_ADV_MAX_ADV_DATA_LEN` macro from `gap.h`.

Reply:
yup. I added a separate commit with new def which also updates few places in code that have "31" hardcoded.